src/riak_cs_copy_object.erl

```diff
@@ -102,15 +104,29 @@ authorize_on_src(RcPid, SrcManifest, RD,
 
 
 %% @doc just kicks up put fsm
--spec start_put_fsm(binary(), binary(), lfs_manifest(),
+-spec new_metadata(lfs_manifest(), #wm_reqdata{}) -> {binary(), [{string(), string()}]}.
+new_metadata(SrcManifest, RD) ->
+    case wrq:get_req_header("x-amz-metadata-directive", RD) of
```

Comment:
The document states that `Constraints: Values other than COPY or REPLACE result in an immediate 400-based error response. You cannot copy an object to itself unless the MetadataDirective header is specified and its value set to REPLACE.` I think we have to handle other values here as 400 or something. And handle `undefined` separately.